components/input/src/input/input.js

```diff
@@ -82,6 +82,23 @@ const styles = css`
         border-color: ${theme.focus};
     }
 
+    input[type='date']::-webkit-inner-spin-button,
+    input[type='date']::-webkit-calendar-picker-indicator,
+    input[type='time']::-webkit-inner-spin-button,
+    input[type='time']::-webkit-calendar-picker-indicator,
+    input[type='datetime-local']::-webkit-inner-spin-button,
+    input[type='datetime-local']::-webkit-calendar-picker-indicator {
+        height: 14px;
+        padding-top: 1px;
+        padding-bottom: 1px;
+    }
+
+    input[type='date']::-webkit-datetime-edit,
+    input[type='datetime-local']::-webkit-datetime-edit,
+    input[type='time']::-webkit-datetime-edit {
+        line-height: 14px;
```

Comment:
I'm assuming this is to style the placeholder text e.g. `mm / dd / yyyy`, if this is not correct you can stop reading because the text below is based on this assumption.

It seems wrong to adjust the placeholder line-height to a value other than the regular line-height (16px). It would be good to fix this issue in another way. If the font and font-size of this placeholder and the regular text are the same (which they should be), then surely something else must be causing the placeholder to take up too much height. I don't think we are tackling the root cause here.

Reply:
I think the root cause is the padding on the placeholder wrapper. Setting that to 0 fixes the height issue too (only tested in the devtools).